app/change_set_persisters/hyrax/change_set_persister.rb

```diff
@@ -3,45 +3,107 @@
 require 'hooks'
 
 module Hyrax
+  # ChangeSetPersisters are used to apply ChangeSets (updates) to resources,
+  # formerly known as curation_concerns, in Hyrax.  A ChangeSetPersister initalizes
+  # with adapters for persisting the metadata and storing the resource.  It should
+  # then be called for the desired operation (save, delete, etc) with the ChangeSet
+  # that contains the resource and changes you wish to make
+  #
+  # @since 3.0.0
   class ChangeSetPersister
     include Hooks
-    define_hooks :before_delete, :after_delete
+    define_hooks :before_save, :after_save, :before_delete, :after_delete
 
     attr_reader :metadata_adapter, :storage_adapter
     delegate :persister, :query_service, to: :metadata_adapter
+
+    # Initalizes the ChangeSetPersister
+    # @param [Valkyrie::MetadataAdapter] metadata_adapter The adapter you wish to
+    #   apply the changes to the resource in
+    # @param [Valkyrie::StorageAdapter] the storage_adapter you wish to store the
+    #   changes to the resource in
     def initialize(metadata_adapter:, storage_adapter:)
       @metadata_adapter = metadata_adapter
       @storage_adapter = storage_adapter
     end
 
+    # Saves all changes in the ChangeSet to the resource contained in the ChangeSet
+    # using the metadata_adapter and storage_adapter
+    # @param change_set Any change_set created by Hyrax::DynamicChangeSet
+    # @return the saved resource, such as GenericWork, that is subclassed from Valkyrie::Resource
     def save(change_set:)
-      persister.save(resource: change_set.resource)
+      run_hook(:before_save, change_set: change_set)
+      resource = persister.save(resource: change_set.resource)
+      run_hook(:after_save, change_set: change_set)
+      query_service.find_by(id: resource.id)
     end
 
+    # Deletes the resource contained in the change_set from the adapters specified
+    # upon initialization of the ChangeSetPersister
+    # @param change_set Any change_set created by Hyrax::DynamicChangeSet
+    # @return [Boolean] true returned if the resource in the change_set was
+    #   succesfully deleted
+    # @return [Boolean] false returned if the resource in the change_set was not deleted
     def delete(change_set:)
-      catch(:abort) do
-        run_hook(:before_delete, change_set: change_set)
-        persister.delete(resource: change_set.resource)
-        run_hook(:after_delete, change_set: change_set)
-      end
+      run_hook(:before_delete, change_set: change_set)
+      persister.delete(resource: change_set.resource)
+      run_hook(:after_delete, change_set: change_set)
     end
 
+    # Calls save on all change_sets in the array passed in
+    # @param [Array] change_sets An array of any change_set created by Hyrax::DynamicChangeSet
+    # @return [Array] an array of the saved resources, such as GenericWork,
+    #  that is subclassed from Valkyrie::Resource
     def save_all(change_sets:)
+      objects = []
```

Comment:
Do we need this accumulator? Can't we just return the result of the `map`?

Reply:
Removed